### PR TITLE
Random number generation should be consistent based on the simulation configuration

### LINKED ISSUE #22: Add a seed value for all simulation randomization for consistent output
Add a randomization seed value in simulation environment configuration for consistent output

### PR DESCRIPTION
**Pull request by tumayt**
Random number generation should be consistent based on the simulation configuration.

All noise generation is based on randomness in simulation environment which can generate different output without a seed. value.

fixes #22

### CODE REVIEW

**Comment by tumayt** [comment]
I really wish to add the parameter to environment node in the config jSON file. But it seems there is a design decision to use the name of the node names as type of the environment to initialize. So I moved the random seed configuration to up one level.

``` js 
"environment": {
  "NoisySharedSubgraph": {
      "noise": true
  }
},
"random_seed":42
```

File: simulation/controller.py [hunk 1]
```diff
@@ -18,6 +20,11 @@ def init_simulation(parser: argparse.ArgumentParser):
         # Load the configuration.
         config = json.loads(f.read())
 
+    # Check for randomization seed value and set
+    if "random_seed" in config:
+        seed_value = config["random_seed"]
+        np.random.seed(seed_value)
```
Comment (tkornuta-semiotic) [inline simulation/controller.py]: please set in here also seed for pytorch - so the agent's params will be initialized also from that seed

Comment (tkornuta-semiotic) [reply simulation/controller.py]: https://vandurajan91.medium.com/random-seeds-and-reproducible-results-in-pytorch-211620301eba

Comment (tumayt) [reply simulation/controller.py]: https://github.com/semiotic-ai/autoagora-agents/pull/23/commits/ee1bef85685ca2fba4b65dae3df67a548fdaa484

**Review by aasseman — CHANGES_REQUESTED** [review]
Great! Could you write a simple Unit Test that would confirm that experiments are reproducible?

**Comment by tumayt** [comment]
I couldn't find any agent using `pytorch` randomization.

**Review by aasseman — APPROVED** [review]
LGTM!

**Pull request merged**